tsl/src/bgw_policy/job.c

```diff
@@ -49,12 +50,12 @@
 #define REORDER_SKIP_RECENT_DIM_SLICES_N 3
 
 static void
-enable_fast_restart(BgwJob *job, const char *job_name)
+enable_fast_restart(int32 job_id, const char *job_name)
 {
-	BgwJobStat *job_stat = ts_bgw_job_stat_find(job->fd.id);
+	BgwJobStat *job_stat = ts_bgw_job_stat_find(job_id);
 
-	ts_bgw_job_stat_set_next_start(job, job_stat->fd.last_start);
-	elog(LOG, "the %s job is scheduled to run again immediately", job_name);
+	ts_bgw_job_stat_set_next_start(job_id, job_stat->fd.last_start);
+	elog(DEBUG1, "the %s job is scheduled to run again immediately", job_name);
```

Comment:
i dont see the value in this message but i think fast restart is wrong approach anyway and the job should just do more in current process instead of rescheduling itself (in separate transactions of course)

Reply:
The advantage of this approach is 1 long-running job will not starve other jobs as the work gets split and scheduled as multiple jobs.

Reply:
i'm  aware but it messes with schedule_interval and you still have max_runtime to control starvation

Reply:
it would allow making consistent start times slightly easier but not something for this PR though